resources/scripts/dag-user-comparator.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+# Check if the correct number of arguments is provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <source_server> <target_server>"
+    exit 1
+fi
+
+# Function to check if a command is available
+check_command() {
+    if ! command -v "$1" &> /dev/null; then
+        echo "Error: $1 is not installed."
+        case "$1" in
+            curl)
+                echo "Install curl using:"
+                echo "  sudo apt-get install curl    # Ubuntu/Debian"
+                echo "  sudo dnf install curl        # Fedora"
+                echo "  brew install curl            # MacOS"
+                ;;
+            jq)
+                echo "Install jq using:"
+                echo "  sudo apt-get install jq      # Ubuntu/Debian"
+                echo "  sudo dnf install jq          # Fedora"
+                echo "  brew install jq              # MacOS"
+                ;;
+            rg)
+                echo "Install ripgrep using:"
+                echo "  sudo apt-get install ripgrep # Ubuntu/Debian"
+                echo "  sudo dnf install ripgrep     # Fedora"
+                echo "  brew install ripgrep         # MacOS"
+                ;;
+        esac
+        exit 1
+    fi
+}
+
+# Check for required commands
+check_command curl
+check_command jq
+check_command rg
+
+# Set PATH variable to ensure the script can find curl, jq, and rg
+export PATH=$PATH:/usr/bin:/usr/local/bin
+
+# Server addresses from arguments
+SOURCE_SERVER="$1"
+TARGET_SERVER="$2"
+
+# Fixed path
+FIXED_PATH="/beta-rewards"
+
+# Construct URLs
+SOURCE_URL="http://$SOURCE_SERVER$FIXED_PATH"
+TARGET_URL="http://$TARGET_SERVER$FIXED_PATH"
+
+# Fetch the JSON data from the source URL
+echo "Fetching data from $SOURCE_URL"
+source_json=$(curl -s $SOURCE_URL)
+
+# Fetch the JSON data from the target URL
+echo "Fetching data from $TARGET_URL"
+target_json=$(curl -s $TARGET_URL)
+
+# Parse the JSON data to extract names, skipping unknown participants
```

Comment:
Not sure if we need to remove the `unknown participants`?
At least on `dags-service` I was asked to keep them (incoming change)

Reply:
We cannot really add them. The added users are about decrypting these entries. So it doesn't amke sense to be adding this to the DAG.